### PR TITLE
[CPDLP-1965] Add validation to has_passed field for npq completed declarations

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-1965

### Changes proposed in this pull request

* Added validation for declarations endpoint to validate `has_passed` when unsupported values are provided.

### CODE REVIEW

File: spec/services/record_declaration_spec.rb [hunk 1]
```diff
@@ -345,6 +345,15 @@
         end
       end
 
+      context "has_passed is invalid text" do
+        let(:has_passed) { "no_supported" }
+
+        it "returns error" do
+          expect(service).to be_invalid
+          expect(service.errors.messages_for(:has_passed)).to eq(["The property '#/has_passed' must be true or false"])
```
Comment (peteryates) [inline spec/services/record_declaration_spec.rb:353]: Are we worried about case here? In the ticket it says `TRUE` / `FALSE`.

**Review by peteryates — COMMENTED** [review]
Looks good over all, one comment on checking the case of requests.

File: spec/services/record_declaration_spec.rb (hunk 1 above)
Comment (peteryates) [reply spec/services/record_declaration_spec.rb:353]: Additionally if we change this spec group to `type: :model` we can check the validations with:

```
expect(service).to allow_value("true", "false").for(:has_passed).with_message("The property...")
```

which might be a bit more concise.

Comment (mooktakim) [reply spec/services/record_declaration_spec.rb:353]: @peteryates 
Added fix for capitalisation.

Comment (mooktakim) [reply spec/services/record_declaration_spec.rb:353]: Speaking with @cwrw, I've changed back to only accepting lowercase to stay consistent with the rest of the API. We'll update the guidance to reflect this.

**Review by cwrw — APPROVED** [review]
👍 , might need to close/reopen not sure whats going on with the review app 🤔

**Pull request closed without merging**